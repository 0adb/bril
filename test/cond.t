A simple program with conditionals:

  $ ts2bril < $TESTDIR/cond.ts | brili
  2

Round-trip it through the text format:

  $ ts2bril < $TESTDIR/cond.ts | python $TESTDIR/../bril-txt/bril2txt.py | python $TESTDIR/../bril-txt/txt2bril.py | brili
  2
