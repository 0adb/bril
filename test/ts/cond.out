main {
  %0 = const 5
  value = id %0
  %1 = id value
  %2 = const 3
  %3 = add %1 %2
  %4 = const 6
  %5 = lt %3 %4
  br %5 then else
then:
  %6 = const 1
  print %6
  %7 = const 0
  jmp endif
else:
  %8 = const 2
  print %8
  %9 = const 0
endif:
}
